pkg/apis/core/validation/validation.go

```diff
@@ -4393,6 +4393,22 @@ func validatePodResourceConsistency(spec *core.PodSpec, fldPath *field.Path) fie
 		}
 	}
 
+	// Pod-level hugepage limits must be >= aggregate limits of all containers in a pod.
```

Comment:
It might be worth adding a summarized comment for why it is not allowed to set pod-level huge pages limits to be greater than aggregated container-level huge pages limits and why huge pages overcommitment is not allowed even with pod-level resources feature. 

We will also have to update the documentation & KEP with this information.